controller/main.go

```diff
@@ -74,16 +75,26 @@ func (c *controller) SetBalancer(l log.Logger, name string, svcRo *v1.Service, _
 	// a reason.
 	svc := svcRo.DeepCopy()
 	syncStateRes := controllers.SyncStateSuccess
+
+	var prevIPs []net.IP
```

Comment:
The current logic does that. If the IPs are erased, it forces a retry all. What I want to do here is to prevent a unnecessary retry all if the previous IP is not valid anymore. It was impossible to have this scenario before because of the logic in setPools, but now it is.

Now, when the service finds another IP, setBalancer does not force a retry all. This is the case in the current logic as well. So, the comparison of IPs is something we could do to improve the code since in theory, the previous ip can be allocated somewhere else. But note that this is out of scope of this PR.

Reply:
I understand that. I am talking about untangling the code because now we have wasAllocated and prevIPs. prevIPs already contains the information that hte service wasAllocated, so we don't need it anymroe. We can  make the logic only on prevIPs + PoolForIP

Reply:
ok, I see what you mean. Will change it. Thanks.